### PR TITLE
[PULL REQUEST] Add capability to write GEOS-Chem Classic timer output to a JSON file

### PR DESCRIPTION
**Pull request by yantosca**
This update adds modifications to `GeosUtil/timers_mod.F90` to output the GEOS-Chem Classic timers to a JSON-format file named `gcclassic_timers.json`.  This will make the timer output easier to process in the new benchmarking infrastructure.  The JSON can be represented as a string and saved into the new benchmarking database.

I confirmed that the `gcclassic_timers.json` is valid JSON output by running it through an online JSON linter (https://jsonlint.com/).

### CODE REVIEW

File: GeosUtil/timers_mod.F90 [hunk 1]
```diff
@@ -977,11 +987,12 @@ END FUNCTION Timer_TheTime
 !\\
 ! !INTERFACE:
 !
-  SUBROUTINE Timer_TimePrint( SlotNumber )
+  SUBROUTINE Timer_TimePrint( SlotNumber, Lun_JSON )
```
Comment (laestrada) [inline GeosUtil/timers_mod.F90:990]: Are there conventions in fortran about using different cases for variables? I know it is case insensitive, but I see `lun_json`, `Lun_JSON`, and `lun_JSON`.  I am still not used to it 🙃

**Review by msulprizio — APPROVED** [review]
These updates look good to me as well.

File: GeosUtil/timers_mod.F90 [hunk 2]
```diff
@@ -594,76 +596,77 @@ SUBROUTINE Timer_Sum_Loop( TimerName, RC )
...
+!! !REVISION HISTORY:
+!!  24 Jul 2015 - M. Yannetti - Initial version.
+!!  See https://github.com/geoschem/geos-chem for complete history
+!!EOP
+!!------------------------------------------------------------------------------
+!!BOC
+!!
+!! !LOCAL VARIABLES:
+!!
+!    ! Scalars
```
[the remark this answers is not in the document]
Comment (laestrada) [reply GeosUtil/timers_mod.F90:639]: Either way sounds fine to me. Just wanted to make sure you had a plan and that this wouldn't be forever forgotten.

**Comment by lizziel** [comment]
@yantosca, could you post a sample output file to give a preview?

File: GeosUtil/timers_mod.F90 (hunk 1 above)
Comment (yantosca) [reply GeosUtil/timers_mod.F90:990]: Good point @laestrada and @msulprizio.  I'll fix that.

**Comment by yantosca** [comment]
> @yantosca, could you post a sample output file to give a preview?

Hi @lizziel, here is what the output looks like: [gcclassic_timers.json.txt](https://github.com/geoschem/geos-chem/files/8035602/gcclassic_timers.json.txt)

File: GeosUtil/timers_mod.F90 (hunk 1 above)
Comment (yantosca) [reply GeosUtil/timers_mod.F90]: See commit 17583e387ddfeed56d17724912fb9be3a9e53993.  I now use camel case for the relevant JSON variables as this matches the convention used in `timers_mod.F90`.

File: GeosUtil/timers_mod.F90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (yantosca) [reply GeosUtil/timers_mod.F90]: @laestrada, I have restored Timer_Print and have made LunJson an optional argument where it needs to be.  See commit 5115c3975595824494a86df205be9181219c8b4d.

[the remark this answers is not in the document]
Comment (laestrada) [reply GeosUtil/timers_mod.F90]: Thanks!

**Comment by yantosca** [comment]
GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

Number of execution tests: 3
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_fullchem_benchmark_merra2_c48...............Execute Simulation.....PASS
gchp_fullchem_standard_merra2_c24................Execute Simulation.....PASS
gchp_TransportTracers_geosfp_c24.................Execute Simulation.....PASS
 
Summary of execution test results:
------------------------------------------------------------------------------
Execution tests passed:        3
Execution tests failed:        0
Execution tests not completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
All GCClassic integration tests passed:
```console
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 48
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Pull request merged**